pandas/tools/merge.py

```diff
@@ -1626,7 +1626,11 @@ def __init__(self, objs, axis=0, join='outer', join_axes=None,
                 clean_objs.append(v)
             objs = clean_objs
             name = getattr(keys, 'name', None)
-            keys = Index(clean_keys, name=name)
+            # GH 14848
+            # Don't pass name when creating index (# GH 14252)
+            # So that if keys are tuples, name isn't checked
```

Comment:
@jreback Just to be clear, are you suggesting that the following code should raise exceptions on both calls to ``Index``? (Right now, the second call succeeds, the first raises)
```
tl = [(0,0,1),(0,1,0),(1,0,0)]
i1 = pd.Index(tl, name='a')
i1 = pd.Index(tl, name='abc')
```
If an exception should be raised in _both_ cases, should I create a separate issue for that (and work on a fix as a separate pull request)? (Note - I think I have a fix for the current issue that is clean that doesn't require modifying `Index`).